src/main/scala/edu/cmu/cs/obsidian/typecheck/Checker.scala

```diff
@@ -1853,22 +1853,27 @@ private def checkStatement(
 
         checkForUnusedStateInitializers(outputContext)
 
-        if (tx.retType.isDefined & !hasReturnStatement(tx, tx.body)) {
-            val ast =
-                if (tx.body.length > 0) {
-                    tx.body.last
-                }
-                else {
-                    tx
+        lexicallyInsideOf.contract match {
+            case ObsidianContractImpl(modifiers, name, declarations, transitions, isInterface, sp) =>
+                // Don't need to check interface methods to make sure they return
+                // Use & instead of && to ensure that all conditions evaluate (hasReturn checks for multiple things)
```

Comment:
Interesting, but it might be better to just change the order so hasReturnStatement goes first. Then you can use &&.

Reply:
That's how it was originally, so I just left it there (after changing it to `&&`, having thought it was a typo, and subsequently spending some time debugging it).

Reply:
I see. Still, it might be worth changing, just for uniformity and simplicity.